### ISSUE TITLE
Zed Mono Font: colon glyph vertical movement

### ISSUE THREAD

**Issue by notpeter** [zed-industries/zed#13060]
### Discussed in https://github.com/zed-industries/zed/discussions/10527

<sup>Originally posted by **notpeter** April 14, 2024</sup>
While coding in Rust I observed an odd behavior where the colon glyphs dance as you type them.

To reproduce type out the following:
```
fn family() -> String {
    String::from(std::consts::FAMILY)
}
```

![colon](https://github.com/zed-industries/zed/assets/145113/b88d1637-7b45-4913-b834-e4f8c4966067)

The colon(s) move twice: First, when you type the second colon (horizontally) and then when you type a letter after (vertically). No idea why the vertical movement only happens inside parens.

The [Iosevka customizer](https://typeof.net/Iosevka/customizer) used to generate [zed-fonts](https://github.com/zed-industries/zed-fonts) doesn't currently support defining variants for the ':' character, but I [opened an issue](https://github.com/be5invis/Iosevka/issues/2321) to see if they might in the future. In writing that up I also discovered the `::` sometimes dances up or dances down or just stays put depending on context. Fun!
![colon-colon2](https://github.com/zed-industries/zed/assets/145113/694bb95d-dc82-412c-9624-afa63bf465d7)

**Comment by Moshyfawn** [comment]
I'm pretty sure this has been reported somewhere, but for some reason I can't seem to find it 🙈

**Comment by someone13574** [comment]
Is it a ligature?

**Comment by notpeter** [comment]
Yes, multiple. I'm fine with the double colon `::` ligature (first colon moves horizontally to make `::` versus a solo `:`) the issue is that there are (at least) two distinct `::` ligatures with different vertical heights.  Which is used depends on the character that follows the `::`  so `::A` or `::` (end of line) is lower than `::<` or `::)`.

Assuming a line where █ is the cursor position:
  - `mem::size_of::█` -> `mem::size_of::<█` -- double colon moves up.
  - `String::from(mem::size_of::█)` -> `String::from(mem::size_of::<█)` -- double colon doesn't move
  - `String::from(mem::█)` -> `String::from(mem::s█)` -- double colon moves down
